### ISSUE TITLE
Feature request: Code blocks in market description

### ISSUE THREAD

**Issue by maltekliemann** [zeitgeistpm/ui#966]
We're slowly progressing towards trying to have an API call for resolving a market. Often enough, these API calls are just web queries, but sometimes we may want to add some code to the market description. Is that possible without too much fuss?

What are the risks here? Permissionless markets posting unsafe code comes to mind...

**Comment by Robiquet** [comment]
Are you talking about adding code snippets to market description like this?:

```import { useState } from "react";```

Not sure I understand the use case

**Comment by maltekliemann** [comment]
Yeah, that's what I mean. Let's say I want to add an API call to the market description like this: https://github.com/maltekliemann/misc-scripts/blob/master/usdc.py. I guess we could just add the link to the script, but maybe a code block like below would be nicer?

```python
from datetime import datetime
import pytz

import krakenex
import pykrakenapi

kraken = pykrakenapi.KrakenAPI(krakenex.API())
time = datetime(2023, 4, 1, 0, 0, tzinfo=pytz.utc)
timestamp = datetime.timestamp(time)
data_frame, _ = kraken.get_ohlc_data("USDCUSD", interval=60 * 24, since=timestamp - 1)
result = data_frame.loc[data_frame["time"] == timestamp]["close"]
print(result)
```

**Comment by Robiquet** [comment]
I think it would be quite easy to add code formatting with the text editing library that we're using. Though I'm not sure if this information is that relevant to most users, so the link to GitHub could be better

**Comment by mk26071980** [comment]
Not sure, if this needs design. Why don't we use brackets <> on the text component to turn on the coding option like on githhub? 
<img width="870" alt="image" src="https://github.com/zeitgeistpm/ui/assets/80706590/8906cc52-07cb-4dc8-81a7-37326fe4d664">
